### DISCUSSION TITLE
Allow miners to configure the minimum/maximum deal durations they're willing to accept

### DISCUSSION THREAD

**Discussion by placer14** [filecoin-project/lotus#7076] [Ideas] [answered]
@Meatball13 had an idea in [a separate related issue](https://github.com/filecoin-project/lotus/issues/4760#issuecomment-896406866):

> I'd like to throw in the suggestion that it'd be nice to allow miners to set a minimum/maximum seal duration they'll accept for deals as well. Right now with lotus-miner storage-deals set-ask you can set prices and min/max piece size, but you can't set Expiry. Maybe some miners only want to hold short term deals, while others want longer term deals. This could be especially useful if say a miner knew that in '3 months' they were going to move their farm or need to take it down or something, if they could set it so, "I won't take any deals that need hosting out longer than this date", that'd be good.

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment, marked as answer]
this is possible via external deal filters

**Reply by placer14** [reply]
Do you have some docs or info that your response could link to?

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
https://docs.filecoin.io/mine/lotus/miner-configuration/#using-filters-for-fine-grained-storage-and-retrieval-deal-acceptance

**Comment by Meatball13** [comment]
This works, in addition to having the capability built into CID Gravity.  We can close this out.
